specs/derivation.md

```diff
@@ -515,22 +515,45 @@ The field elements are encoded as big-endian integers (`KZG_ENDIANNESS = big`).
 
 To save computational overhead, only `254` bits per field element are used for rollup data.
 
-Data is encoded, 4 field elements, `127` bytes of rollup data, at a time:
+`127` bytes of application-layer rollup data is encoded at a time, into 4 adjacent field elements of the blob:
 
-```text
-# Allocate 4 field elements (big-endian).
-# Read 31 bytes into the low 31 bytes of each.
-<----- element 0 -----><----- element 1 -----><----- element 2 -----><----- element 3 ----->
-| byte A |  read(31)  || byte B |  read(31)  || byte C |  read(31)  || byte D |  read(31)  |
-# Read 3 more bytes after the above.
+```python
+# read(N): read N bytes from the application-layer rollup-data.
```

Comment:
writing appends and keeps an offset internal to its implementation but read doesn't take an offset so its a little unclear

Reply:
clarified the read function :+1: